app/models/form/sales/pages/person.rb

```diff
@@ -0,0 +1,18 @@
+class Form::Sales::Pages::Person < ::Form::Page
+  def initialize(id, hsh, subsection, person_index:)
+    super(id, hsh, subsection)
+    @person_index = person_index
+  end
+
+  def person_display_number
+    joint_purchase? ? @person_index - 2 : @person_index - 1
+  end
+
+  def joint_purchase?
+    id.include?("_joint_purchase")
+  end
+
+  def details_known_question_id
+    "details_known_#{person_display_number}"
+  end
```

Comment:
```suggestion
  def field_for_person(field)
    [field, person_display_number].join
  end
  
  def field_for_person_index(field)
      [person_index, field].join
  end
```

Was there a motivation for not making it flexible for any other field?

Reply:
Oh, yes, sorry, that's fine. Have a look now. I've added another method for that tho, because details_known uses display number instead of the database field. That's because details_known actually starts counting from 1 in all cases (because we only ask details known about people and not buyers, no matter how many of them we have)